app/(pages)/(OrderSystem)/order/date/page.js

```diff
@@ -117,66 +130,101 @@ export default function TimePicker() {
     );
     setSelectedTime(dateTime);
     setShowTimeModal(false);
+    console.log(disabledTimes);
   };
 
-  // Update email state and displayEmail state when input value changes
   const handleEmailChange = (e) => {
     const newEmail = e.target.value;
-    setEmail(newEmail);
+
+    setUserEmail(newEmail);
     setDisplayEmail(newEmail);
   };
 
+  function disableTimes(day) {
+    const date = ("selected date", moment(day).format("YYYY-MM-DD"));
+
+    const StoredData = Object.keys(localStorage).map((key) => {
+      let parsedData = null;
+      try {
+        parsedData = JSON.parse(localStorage.getItem(key));
+      } catch (error) {
+        console.error(`Error parsing data for key ${key}: ${error}`);
+      }
+      return parsedData;
+    });
+
+    const validStoredData = StoredData.filter((data) => {
+      return data && data.date === date && data.time;
+    });
+
+    const foundTimes = validStoredData
+      .map((data) => (data.date === date ? data.time : undefined)) // map to time or undefined
+      .filter((time) => time !== undefined); // filter out undefined values
+
+    setDisabledTimes(foundTimes);
+  }
+
```

Comment:
The code looks fine, and it works, though you could simplify it a little:

In your function, you could change it so you instead convert localStorage to an array of objects. Then, filter the data to keep only the valid data with the correct date and time, and then at the end, map the filtered data to get an array of disabled times.

I once wrote a similar code to a project I worked on where I did something like this:
`function isTimeSlotTaken(date, time) {
  const allStoredData = Object.keys(localStorage).map(key => JSON.parse(localStorage.getItem(key)));
  return allStoredData.some(savedData => savedData?.date?.date === date && savedData?.date?.time === time);
}`


So in your instance, you could potentially replace it with something like this:

`function disableTimes(day) {
  const date = moment(day).format("YYYY-MM-DD");
  const allStoredData = Object.keys(localStorage).map((key) => JSON.parse(localStorage.getItem(key)));
  const foundTimes = allStoredData
    .filter((savedData) => savedData?.date?.date === date && savedData?.date?.time)
    .map((savedData) => savedData.date.time);
  setDisabledTimes(foundTimes);
}`

You could always try and write something yourself, based on the logic from the code I wrote in another project, or implement the code I wrote in your existing one.🚀

Reply:
Thank you for the recommendation, and for trying to find a better way to write this function. I decided to write it this way because not everything saved in the local storage is a valid JSON and I kept getting errors. This way, I am getting only the valid data.